tools/distpartitioning/multi_dev_init.py

```diff
@@ -0,0 +1,513 @@
+        as column data. This information is read from the edges.txt file.
+    """
+    input_list = []
+    send_sizes = []
+    recv_sizes = []
+    start = timer()
+    for i in np.arange(world_size):
+        send_idx = (edge_data[constants.OWNER_PROCESS] == i)
+        send_idx = send_idx.reshape(edge_data[constants.GLOBAL_SRC_ID].shape[0])
+        filt_data = np.column_stack((edge_data[constants.GLOBAL_SRC_ID][send_idx == 1], \
+                                    edge_data[constants.GLOBAL_DST_ID][send_idx == 1], \
+                                    edge_data[constants.GLOBAL_TYPE_EID][send_idx == 1], \
+                                    edge_data[constants.ETYPE_ID][send_idx == 1], \
+                                    edge_data[constants.GLOBAL_EID][send_idx == 1]))
+        if(filt_data.shape[0] <= 0):
+            input_list.append(torch.empty((0,), dtype=torch.int64))
+            send_sizes.append(torch.empty((0,), dtype=torch.int64))
+        else:
+            input_list.append(torch.from_numpy(filt_data))
+            send_sizes.append(torch.tensor(filt_data.shape, dtype=torch.int64))
+        recv_sizes.append(torch.zeros((2,), dtype=torch.int64))
+    end = timer()
+    
+    dist.barrier ()
+    start = timer()
+    alltoall_cpu(rank, world_size, recv_sizes, send_sizes)
+    output_list = []
+    for s in recv_sizes: 
+        output_list.append(torch.zeros(s.tolist(), dtype=torch.int64))
+
+    dist.barrier ()
+    alltoallv_cpu(rank, world_size, output_list, input_list)
+    end = timer()
+    print('[Rank: ', rank, '] Time to send/rcv edge data: ', timedelta(seconds=end-start))
+
+    #Replace the values of the edge_data, with the received data from all the other processes.
+    rcvd_edge_data = torch.cat(output_list).numpy()
+    edge_data[constants.GLOBAL_SRC_ID] = rcvd_edge_data[:,0]
+    edge_data[constants.GLOBAL_DST_ID] = rcvd_edge_data[:,1]
+    edge_data[constants.GLOBAL_TYPE_EID] = rcvd_edge_data[:,2]
+    edge_data[constants.ETYPE_ID] = rcvd_edge_data[:,3]
+    edge_data[constants.GLOBAL_EID] = rcvd_edge_data[:,4]
+    edge_data.pop(constants.OWNER_PROCESS)
+
+
+def exchange_node_features(rank, world_size, node_data, node_features, ntypes_map, \
+        ntypes_nid_map, ntype_id_count, node_part_ids):
+    """
+    This function is used to shuffle node features so that each process will receive
+    all the node features whose corresponding nodes are owned by the same process. 
+    The mapping procedure to identify the owner process is not straight forward. The
+    following steps are used to identify the owner processes for the locally read node-
+    features. 
+    a. Compute the global_nids for the locally read node features. Here metadata json file
+        is used to identify the corresponding global_nids. Please note that initial graph input
+        nodes.txt files are sorted based on node_types. 
+    b. Using global_nids and metis partitions owner processes can be easily identified. 
```

Comment:
Can you give the details of how to use `global_nids` and `metis partitions` to identify owner processes?